components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/GroupResource.java

```diff
@@ -333,29 +462,144 @@ private Response processRequest(final Map<String, String> requestAttributes) {
+                    e), encoder);
+        }
+    }
+
+    /**
+     * Get group display name from the groupId.
+     *
+     * @param groupId SCIM group id.
+     * @return Display name of the group.
+     * @throws JSONException thrown when an error occurred when getting displayName from JSON response.
+     */
+    private String getGroupName(String groupId, SCIMUserManager userManager, GroupResourceManager groupResourceManager,
+                                String excludeAttributes) throws JSONException {
+
+        String includeAttributes = SCIMConstants.GroupSchemaConstants.DISPLAY_NAME;
+        SCIMResponse scimResponse = groupResourceManager.get(groupId, userManager, includeAttributes, excludeAttributes);
+        JSONObject responseMessage = new JSONObject(scimResponse.getResponseMessage());
+        try {
+            return (String) (responseMessage).get(SCIMConstants.GroupSchemaConstants.DISPLAY_NAME);
+        } catch (JSONException e) {
+            throw new JSONException("An error occurred when extracting DisplayName from group: " + groupId);
+        }
+    }
+
+    /**
+     * Decode patch operation resource string and get the permissions.
+     *
+     * @param jsonResourceString string that should decode.
+     * @return Map of permissions to add and remove.
+     * @throws BadRequestException
+     */
+    private HashMap<String, List<String>> decodePatchOperation(String jsonResourceString) throws BadRequestException {
+
+        JSONDecoder decode = new JSONDecoder();
+        ArrayList<PatchOperation> listOperations;
+        List<String> permissionsToAdd;
+        List<String> permissionsToRemove;
+        HashMap<String, List<String>> permissionMap = new HashMap<>();
+        try {
+            listOperations = decode.decodeRequest(jsonResourceString);
+
+            if (!listOperations.isEmpty()) {
+                for (PatchOperation op : listOperations) {
+                    if ((SCIMProviderConstants.ADD).equals(op.getOperation())) {
+                        JSONArray permissions = new JSONArray(op.getValues().toString());
+                        permissionsToAdd = IntStream.range(0, permissions.length()).mapToObj(permissions::getString)
+                                .collect(Collectors.toList());
+                        permissionMap.put(SCIMProviderConstants.ADD, permissionsToAdd);
+                    } else if (SCIMProviderConstants.REMOVE.equals(op.getOperation())) {
+                        JSONArray permissions = new JSONArray(op.getValues().toString());
+                        permissionsToRemove = IntStream.range(0, permissions.length()).mapToObj(permissions::getString)
+                                .collect(Collectors.toList());
+                        permissionMap.put(SCIMProviderConstants.REMOVE, permissionsToRemove);
+                    }
+                }
+            }
+            return permissionMap;
+        } catch (BadRequestException e) {
```

Comment:
This hides the exact error **e**

Don't need to catch BadRequestException since it is thrown..